src/components/SideNavigation/SideNavigationLink/SideNavigationLink.tsx

```diff
@@ -16,22 +16,30 @@ export type Props<L = LinkDefaultElement> = Omit<
    * @default a
    */
   component?: SideNavigationBaseProps<L>["component"];
+  /**
+   * A ref to pass to the link element.
+   */
+  forwardRef?: React.Ref<HTMLAnchorElement> | null;
```

Comment:
The element won't always be an anchor (if someone has passed the `component` prop) so this might need a new generic that could be set to `HTMLAnchorElement` by default.